lib/database/adapters/LhcPeriodAdapter.js

```diff
@@ -29,8 +29,15 @@ class LhcPeriodAdapter {
      * @param {SequelizeLhcPeriod} databaseObject Object to convert.
      * @returns {LhcPeriod} Converted entity object.
      */
-    toEntity({ id, name }) {
-        return { id, name };
+    toEntity(databaseObject) {
+        const { id, name } = databaseObject;
+        const { runs, dataPasses } = databaseObject;
+        const runsCount = runs ? runs[0]?.get('count') ?? 0 : null;
+        const distinctEnergies = runs ? runs[0]?.get('distinctEnergies')?.split(',').map((energy) => Number(energy)) ?? [] : null;
+        const beamType = runs ? runs[0]?.get('beamType')?.split(',') ?? [] : null;
+
+        const dataPassesCount = dataPasses ? dataPasses[0]?.get('count') ?? 0 : null;
```

Comment:
This is a VERY hard to read code. Ternary should only be used when both branches are simple, which is not the case here. Plus, when you have 3 times the same ternary, you should extract an if. For example:
```js
const runsCount = runs ? runs[0]?.get('count') ?? 0 : null;
const distinctEnergies = runs ? runs[0]?.get('distinctEnergies')?.split(',').map((energy) => Number(energy)) ?? [] : null;
const beamType = runs ? runs[0]?.get('beamType')?.split(',') ?? [] : null;
```

Would be way more clear this way:

```js
let runsCount = null;
let distinctEnergies = null;
let beamType = null;
if (runs?.length) {
    runsCount = runs[0].get('count') ?? 0;
    distinctEnergies = (runs[0].get('distinctEnergies') ?? '').split(',').map(parseInt);
    beamType = (runs[0].get('beamType') ?? '').split(',');
}
```

We usually use parseInt and not number, and if you are not explicitly needing Number specificity, you should try to stick to parseInt.

Reply:
does not apply anymore

Reply:
> We usually use parseInt and not number, and if you are not explicitly needing Number specificity, you should try to stick to parseInt

This still applies